extract.sh

```diff
@@ -1,59 +1,88 @@
 #!/bin/bash
-set -evo pipefail
+set -eo pipefail
 
-IMG_TOPICS=("/basler/pylon_camera_node0/image_raw" "/basler/pylon_camera_node1/image_raw")
-PCD_TOPICS=("/azure/points2" "/velodyne/velodyne_points")
-IMU_TOPICS=("/mcu/mcu_imu")
-TEMP_TOPICS=("/mcu/mcu_imu_temp")
-TIME_REF_TOPICS=("/mcu/cameras_ts" "/mcu/lidar_ts")
+SMARTPHONE_VIDEO_DIR="_s10_video_frame"
 
+# Import configuration values
+source extract.conf
 # Activate virtual environment
-source venv/bin/activate;
+source venv/bin/activate
 # Launch roscore in background
 roscore &
 
-for BAG in "${@}";
-  do
-  DATA_DIR="$(basename "$BAG" .bag)"
+BAG=$1
+SMARTPHONE_VIDEO_PATH=$2
 
-  # Create a subdirectory for extraction
-  rm -rf "$DATA_DIR";
-  mkdir "$DATA_DIR";
+DATA_DIR="$(basename "$BAG" .bag)"
 
-  # Image extraction
-  echo "Image data extraction starting..";
-  python2 extract.py --output "$DATA_DIR"\
-    --type image --path "$BAG" --topics "${IMG_TOPICS[@]}";
+# Create a subdirectory for extraction
+rm -rf "$DATA_DIR"
+mkdir "$DATA_DIR"
+
+# Time reference data extraction
+echo "Time reference data extraction starting.."
+python2 extract.py --output "$DATA_DIR"\
+  --type time_ref --path "$BAG" --topics "${TIME_REF_TOPICS[@]}"
+
+# Image extraction
+echo "Image data extraction starting.."
 
-  # IMU data extraction
-  echo "IMU data extraction starting..";
+if [ ${#IMG_TOPICS[@]} -eq 0 ] ; then
+  echo "No image topics provided"
+else
   python2 extract.py --output "$DATA_DIR"\
-    --type imu --path "$BAG" --topics "${IMU_TOPICS[@]}" --temp "${TEMP_TOPICS[@]}";
+  --type image --path "$BAG" --topics "${IMG_TOPICS[@]}"
+fi
 
-  # Time reference data extraction
-  echo "Time reference data extraction starting..";
+# Depth image extraction
+echo "Image data extraction starting.."
+if [ ${#DEPTH_IMG_TOPICS[@]} -eq 0 ] ; then
+  echo "No image topics provided"
+else
   python2 extract.py --output "$DATA_DIR"\
-    --type time_ref --path "$BAG" --topics "${TIME_REF_TOPICS[@]}";
+    --type depth_img --path "$BAG" --topics "${DEPTH_IMG_TOPICS[@]}"
+  # Depth image timestamps alignment
+  for topic in "${DEPTH_IMG_TOPICS[@]}"
+  do
+    python2 align.py --time_ref_file "./$DATA_DIR"/_mcu_cameras_ts/time_ref.csv\
+     --target_dir "./$DATA_DIR/${topic//\//_}" --align_type ref  --ref_seq 12
+  done
+fi
+
+# Smartphone data alignment
+if [ -z "$1" ]
+  then
+    echo "No smartphone video argument supplied"
+  else
+    # Create target directory, extract video frames
+    rm -rf "./$DATA_DIR/$SMARTPHONE_VIDEO_DIR"
+    mkdir "./$DATA_DIR/$SMARTPHONE_VIDEO_DIR"
+    ffmpeg -i "$SMARTPHONE_VIDEO_PATH" "./$DATA_DIR/$SMARTPHONE_VIDEO_DIR/frame-%d.png"
```

Comment:
What if it doesn't exist? For example, the argument is provided but with incorrect name. Provide a check for this.

Reply:
The same check for .bag file in the beginning.